pkg/apis/serving/v1alpha1/domainmapping_validation.go

```diff
@@ -20,43 +20,58 @@ import (
 	"context"
 	"fmt"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/api/validation"
 	"knative.dev/pkg/apis"
-	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/serving/pkg/apis/serving"
+	v1 "knative.dev/serving/pkg/apis/serving/v1"
 )
 
 // Validate makes sure that DomainMapping is properly configured.
-func (dm *DomainMapping) Validate(_ context.Context) *apis.FieldError {
-	return validateMetadata(&dm.ObjectMeta).ViaField("metadata").
-		Also(dm.validateSpec().ViaField("spec"))
-}
+func (dm *DomainMapping) Validate(ctx context.Context) *apis.FieldError {
+	errs := dm.validateMetadata(ctx).ViaField("metadata")
+
+	ctx = apis.WithinParent(ctx, dm.ObjectMeta)
+	errs = errs.Also(dm.Spec.Validate(apis.WithinSpec(ctx)).ViaField("spec"))
 
-// validateSpec validates the Spec of a DomainMapping.
-func (dm *DomainMapping) validateSpec() (errs *apis.FieldError) {
-	return errs.Also(validateRef(&dm.Spec.Ref, dm.Namespace).ViaField("ref"))
+	return errs
 }
 
-// validateRef validates the Ref field of a DomainMapping.
-func validateRef(ref *duckv1.KReference, ns string) (errs *apis.FieldError) {
-	if ref.Name == "" {
-		errs = errs.Also(apis.ErrMissingField("name"))
+// validateMetadata validates the metadata section of a DomainMapping.
+func (dm *DomainMapping) validateMetadata(ctx context.Context) (errs *apis.FieldError) {
+	if dm.GenerateName != "" {
+		errs = errs.Also(apis.ErrDisallowedFields("GenerateName"))
```

Comment:
```suggestion
		errs = errs.Also(apis.ErrDisallowedFields("generateName"))
```

We use json/yaml casing in these

Reply:
oops, fixed - thanks